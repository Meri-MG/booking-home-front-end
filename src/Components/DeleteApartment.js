import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { removeApartment } from '../Redux/Apartments/Apartments';

const DeleteApartment = () => {
  const apartments = useSelector((state) => state.Apartments.apartments);
  const dispatch = useDispatch();

  return (
    <div className="h-full min-h-screen bg-main-color bg-cover flex flex-col">
      <h2 className="text-3xl text-white text-center font-semibold p-9">Delete Apartments</h2>
      {apartments?.map((apartment) => (
        <article
          key={apartment.apartment}
          className="flex justify-evenly items-center rounded shadow-sm hover:scale-105 hover:shadow-md transition-transform bg-white p-2 mr-8 ml-8 mb-4"
        >
          <h3 className="text-xl font-semibold">{apartment.name}</h3>
          <button
            className="text-center text-white bg-red-500 hover:bg-red-600 active:bg-red-700 transition-colors p-3 rounded-xl font-semibold"
            type="button"
            onClick={() => dispatch(removeApartment(apartment.apartment))}
          >
            Delete Apartment
          </button>
        </article>
      ))}
    </div>
  );
};

export default DeleteApartment;
