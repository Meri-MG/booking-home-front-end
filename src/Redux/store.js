import { applyMiddleware, combineReducers, createStore } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import loginReducer from './Login/Login';
import registerReducer from './Register/Register';
import apartmentsReducer from './Apartments/Apartments';
import apartmentsDetailsReducer from './Details/Details';
import reservationsReducer from './Reserved/Reserved';

const reducers = combineReducers({
  LogIn: loginReducer,
  Register: registerReducer,
  Apartments: apartmentsReducer,
  Details: apartmentsDetailsReducer,
  Reservations: reservationsReducer,
});

const middlewares = [thunk, logger];

const store = createStore(reducers, applyMiddleware(...middlewares));

export default store;
