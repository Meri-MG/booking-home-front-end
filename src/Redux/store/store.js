import { applyMiddleware, combineReducers, createStore } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';

const reducers = combineReducers({
});

const middlewares = [thunk,logger];

const store = createStore(reducers, applyMiddleware(...middlewares));

export default store;
