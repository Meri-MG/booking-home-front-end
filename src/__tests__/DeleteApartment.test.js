import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../Redux/store';
import DeleteApartment from '../Components/DeleteApartment';

it('renders delete apartments page correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <DeleteApartment />
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
