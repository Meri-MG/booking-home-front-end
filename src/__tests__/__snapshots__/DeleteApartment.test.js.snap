// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders delete apartments page correctly 1`] = `
<div
  className="h-full min-h-screen bg-main-color bg-cover flex flex-col"
>
  <h2
    className="text-3xl text-white text-center font-semibold p-9"
  >
    Delete Apartments
  </h2>
</div>
`;
